### ISSUE TITLE
[QUESTION] Problem with installation

### ISSUE THREAD

**Issue by kalylian** [UtrechtUniversity/yoda-ansible#70]
### ansible problem with installation

Hi,

I'm trying to install Yoda following the documentation available at https://utrechtuniversity.github.io/yoda/administration/configuring-yoda.html. Disclaimer, I'm very new at ansible.

So I tried creating a hosts file following the example given in the documentation. That gave me a "Section [yoda:children] includes undefined group" error. I compared the example file with the hosts file in "development/allinone" provided by this git repo and noticed that in the example file in the documentation, the section [yoda:children] contains host names, whereas in development/allinone, it contains the groups of hosts that are defined beneath. I adapted my hosts-file to match the one in development/allinone. The file seems to be parsed. but I'm getting a new error now. I thought I'd check whether I'm on the right track, since I'm currently not following the documentation.

Kind regards

Gudrun

**Comment by stsnel** [comment]
Hello Gudrun,

It appears that the configuration example in our documentation does not work as expected. Thanks for the heads up. We'll review and update it - this issue has been registered in our internal bug tracker as YDA-4241. 

The inventory structure of development/allinone should be right; this is also the structure we're using on our organization's Yoda environments. So you're on the right track. If you wish to share the error message you're getting, perhaps we can help with troubleshooting.

Best,
Sietse

**Comment by kalylian** [comment]
Hi,

thanks for the reply.

I've set up a CentOS VM, installed vagrant, virtualbox and ansible and cloned the git repo.

I've copied allinone/group_vars/allinone.yml to my environment, changed all FQDN's, instance name, environment and version. This is my output when executing ansible-playbook -i environments/production playbook.yml --limit=yoda -K :

```
BECOME password: 

PLAY [localhost] ******************************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY [all] ************************************************************************************************************************************************************************************************

TASK [Retrieve Yoda repository branch] ********************************************************************************************************************************************************************
ok: [yodatest]

TASK [Verify Yoda repository is on correct branch] ********************************************************************************************************************************************************
fatal: [yodatest]: FAILED! => {"msg": "The conditional check 'not ansible_check_mode and yoda_version not in git_branch.stdout and yoda_environment != \"development\"' failed. The error was: error while evaluating conditional (not ansible_check_mode and yoda_version not in git_branch.stdout and yoda_environment != \"development\"): Unable to look up a name or access an attribute in template string ({% if not ansible_check_mode and yoda_version not in git_branch.stdout and yoda_environment != \"development\" %} True {% else %} False {% endif %}).\nMake sure your variable name does not contain invalid characters like '-': coercing to Unicode: need string or buffer, AnsibleUndefined found\n\nThe error appears to be in '/root/yoda/yoda/playbook.yml': line 29, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n    - name: Verify Yoda repository is on correct branch\n      ^ here\n"}

PLAY RECAP ************************************************************************************************************************************************************************************************
yodatest                   : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

Playbook run took 0 days, 0 hours, 0 minutes, 0 seconds
```

I think this is either a problem with my setup or with yoda.yml. Here's the head of the file (the rest is the file from allinone with all FQDN replaced accordingly):

```

:# head -n 20 environments/production/yoda/group_vars/yoda.yml 
---
# copyright Utrecht University

# Ansible configuration
ansible_user: vagrant                             # Administrative user on instance for Ansible
ansible_ssh_private_key_file: vagrant/ssh/vagrant # Path to private key file of administrative user

# Yoda configuration
instance: yoda                                     # Name of Yoda instance, as defined in hosts file
yoda_version: release-1.7                         # Git branch, for example: development or release-1.5
yoda_environment: production                      # Yoda environment: development, testing, acceptance or production
yoda_portal_fqdn: $FQDN                 # Yoda Portal fully qualified domain name (FQDN)
yoda_davrods_fqdn: $FQDN                   # Yoda Davrods WebDAV fully qualified domain name (FQDN)
yoda_davrods_anonymous_enabled: true
yoda_davrods_anonymous_fqdn: $FQDN  # Yoda Davrods anonymous WebDAV fully qualified domain name (FQDN)

default_yoda_schema: default-1                     # Default Yoda metadata scheme: core-0 or default-1
```


Kind regards,

Gudrun

**Comment by kalylian** [comment]
Hi,

I figured it out. I think. I used the command given in the documentation (_ansible-playbook -i environments/production playbook.yml --limit=yoda -K_) and that failed. I changed it to "_ansible-playbook -i environments/production/yoda/ playbook.yml -K_" . Now the playbook runs. Yoda is still not really running, but according to the error messages, that's because I've not configured group_vars/yoda.yml properly. That's kind of expected, since I haven't adapted most of the options.

Kind regards,

Gudrun

**Comment by lwesterhof** [comment]
Hi Gudrun,

From the  environments/production/yoda/group_vars/yoda.yml  configuration it looks you have configured to install release-1.7. The task on which the playbook fails (Verify Yoda repository is on correct branch) will check that you are on the release-1.7 branch instead of the default development branch. A 'git checkout release-1.7' should bring you further!

Best regards,
Lazlo

**Comment by kalylian** [comment]
Hi

I verified that branch and yoda_version (and switched both to development), only got it working when adjusting the command. Anyway, it works now.

My current problem looks like this. I start the playbook, it ends at this point:

> TASK [yoda_rulesets : Ensure metadata schemas are installed] **********************************************************************************************************************************************
> failed: [test-yoda] (item=core-0) => {"ansible_loop_var": "item", "changed": false, "cmd": ["irule", "-F", "/etc/irods/irods-ruleset-uu/tools/install-metadata-schema.r", "*resc=\"irodsResc\"", "*src=\"/etc/irods/irods-ruleset-uu/schemas/\"", "*schema=\"core-0\"", "*category=\"core-0\"", "*update=1"], "delta": "0:00:03.457732", "end": "2021-09-13 14:07:28.528572", "item": "core-0", "msg": "non-zero return code", "rc": 4, "start": "2021-09-13 14:07:25.070840", "stderr": "remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT", "stderr_lines": ["remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT"], "stdout": "Level 0: DEBUG: execMicroService3: error when executing microservice\nline 76, col 2\n\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);\n\t\t^\n\ncaused by: DEBUG: msiDataObjPut: rsDataObjPut failed for 3Ú\u0001, status = -9000", "stdout_lines": ["Level 0: DEBUG: execMicroService3: error when executing microservice", "line 76, col 2", "\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);", "\t\t^", "", "caused by: DEBUG: msiDataObjPut: rsDataObjPut failed for 3Ú\u0001, status = -9000"]}
> failed: [test-yoda] (item=default-0) => {"ansible_loop_var": "item", "changed": false, "cmd": ["irule", "-F", "/etc/irods/irods-ruleset-uu/tools/install-metadata-schema.r", "*resc=\"irodsResc\"", "*src=\"/etc/irods/irods-ruleset-uu/schemas/\"", "*schema=\"default-0\"", "*category=\"default-0\"", "*update=1"], "delta": "0:00:03.414319", "end": "2021-09-13 14:07:32.279551", "item": "default-0", "msg": "non-zero return code", "rc": 4, "start": "2021-09-13 14:07:28.865232", "stderr": "remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT", "stderr_lines": ["remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT"], "stdout": "Level 0: DEBUG: execMicroService3: error when executing microservice\nline 76, col 2\n\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);\n\t\t^\n\ncaused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000", "stdout_lines": ["Level 0: DEBUG: execMicroService3: error when executing microservice", "line 76, col 2", "\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);", "\t\t^", "", "caused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000"]}
> failed: [test-yoda] (item=default-1) => {"ansible_loop_var": "item", "changed": false, "cmd": ["irule", "-F", "/etc/irods/irods-ruleset-uu/tools/install-metadata-schema.r", "*resc=\"irodsResc\"", "*src=\"/etc/irods/irods-ruleset-uu/schemas/\"", "*schema=\"default-1\"", "*category=\"default-1\"", "*update=1"], "delta": "0:00:03.390114", "end": "2021-09-13 14:07:35.977541", "item": "default-1", "msg": "non-zero return code", "rc": 4, "start": "2021-09-13 14:07:32.587427", "stderr": "remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT", "stderr_lines": ["remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT"], "stdout": "Level 0: DEBUG: execMicroService3: error when executing microservice\nline 76, col 2\n\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);\n\t\t^\n\ncaused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000", "stdout_lines": ["Level 0: DEBUG: execMicroService3: error when executing microservice", "line 76, col 2", "\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);", "\t\t^", "", "caused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000"]}
> failed: [test-yoda] (item=hptlab-0) => {"ansible_loop_var": "item", "changed": false, "cmd": ["irule", "-F", "/etc/irods/irods-ruleset-uu/tools/install-metadata-schema.r", "*resc=\"irodsResc\"", "*src=\"/etc/irods/irods-ruleset-uu/schemas/\"", "*schema=\"hptlab-0\"", "*category=\"hptlab-0\"", "*update=1"], "delta": "0:00:06.337774", "end": "2021-09-13 14:07:42.630617", "item": "hptlab-0", "msg": "non-zero return code", "rc": 4, "start": "2021-09-13 14:07:36.292843", "stderr": "remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT", "stderr_lines": ["remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT"], "stdout": "Level 0: DEBUG: execMicroService3: error when executing microservice\nline 76, col 2\n\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);\n\t\t^\n\ncaused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000", "stdout_lines": ["Level 0: DEBUG: execMicroService3: error when executing microservice", "line 76, col 2", "\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);", "\t\t^", "", "caused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000"]}
> failed: [test-yoda] (item=teclab-0) => {"ansible_loop_var": "item", "changed": false, "cmd": ["irule", "-F", "/etc/irods/irods-ruleset-uu/tools/install-metadata-schema.r", "*resc=\"irodsResc\"", "*src=\"/etc/irods/irods-ruleset-uu/schemas/\"", "*schema=\"teclab-0\"", "*category=\"teclab-0\"", "*update=1"], "delta": "0:00:03.340380", "end": "2021-09-13 14:07:46.280943", "item": "teclab-0", "msg": "non-zero return code", "rc": 4, "start": "2021-09-13 14:07:42.940563", "stderr": "remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT", "stderr_lines": ["remote addresses: IP ERROR: rcExecMyRule error.  status = -9000 SYS_EXCEED_CONNECT_CNT"], "stdout": "Level 0: DEBUG: execMicroService3: error when executing microservice\nline 76, col 2\n\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);\n\t\t^\n\ncaused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000", "stdout_lines": ["Level 0: DEBUG: execMicroService3: error when executing microservice", "line 76, col 2", "\t\tmsiDataObjPut(*jsondefault, *resc, \"localPath=*src/*schema/*defaultJsonSchema\", *status);", "\t\t^", "", "caused by: DEBUG: msiDataObjPut: rsDataObjPut failed for HîðL, status = -9000"]}
> 
> PLAY RECAP ************************************************************************************************************************************************************************************************
> localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
> test-yoda                  : ok=222  changed=131  unreachable=0    failed=1    skipped=42   rescued=0    ignored=4   
> 
> Playbook run took 0 days, 0 hours, 12 minutes, 1 seconds


It seems that iRODS isn't working correctly:

> [root@test-yoda:~/yoda/yoda]# su irods
> bash-4.2$ cd
> bash-4.2$ iinit
> Enter your current iRODS password:
> bash-4.2$ echo "Hello world" > test.txt
> bash-4.2$ iput test.txt 
> remote addresses: IP ERROR: putUtil: put error for /tempZone/home/rods/test.txt, status = -9000 status = -9000 SYS_EXCEED_CONNECT_CNT
> 

That looks very broken. Since it's the same error message, it's probably related to the failure above. The weird part is, iput seems to have worked before:
  

> bash-4.2$ tail -n 30 /var/lib/irods/log/setup_log.txt 
> 2021-09-13 12:02:43,246 -   DEBUG -                         lib.py: 129 - Command ['/usr/sbin/irodsServer'] returned with code 0.
> 2021-09-13 12:02:43,247 -   DEBUG -                  controller.py: 102 - Attempting to connect to iRODS server on port 1247. Attempt #1
> 2021-09-13 12:02:44,248 -   DEBUG -                  controller.py: 102 - Attempting to connect to iRODS server on port 1247. Attempt #2
> 2021-09-13 12:02:44,249 -   DEBUG -                  controller.py: 106 - Successfully connected to port 1247.
> 2021-09-13 12:02:44,274 -    INFO -                  controller.py: 123 - Success
> 2021-09-13 12:02:44,275 -    INFO -                 setup_irods.py: 139 - 
> +---------------------+
> | Attempting test put |
> +---------------------+
> 
> 2021-09-13 12:02:44,275 -    INFO -                 setup_irods.py: 145 - Putting the test file into iRODS...
> 2021-09-13 12:02:44,275 -   DEBUG -                         lib.py:  56 - Calling ['iput', '/tmp/tmpml8KccirodsResc'] with options:
> {'shell': False, 'stderr': -1, 'stdout': -1}
> 2021-09-13 12:02:44,463 -   DEBUG -                         lib.py: 129 - Command ['iput', '/tmp/tmpml8KccirodsResc'] returned with code 0.
> 2021-09-13 12:02:44,464 -    INFO -                 setup_irods.py: 148 - Getting the test file from iRODS...
> 2021-09-13 12:02:44,465 -   DEBUG -                         lib.py:  56 - Calling ['iget', 'tmpml8KccirodsResc', '-'] with options:
> {'shell': False, 'stderr': -1, 'stdout': -1}
> 2021-09-13 12:02:44,588 -   DEBUG -                         lib.py: 129 - Command ['iget', 'tmpml8KccirodsResc', '-'] returned with code 0.
> stdout:
>   This is a test file written by the iRODS installation script.
> 2021-09-13 12:02:44,588 -    INFO -                 setup_irods.py: 151 - Removing the test file from iRODS...
> 2021-09-13 12:02:44,589 -   DEBUG -                         lib.py:  56 - Calling ['irm', '-f', 'tmpml8KccirodsResc'] with options:
> {'shell': False, 'stderr': -1, 'stdout': -1}
> 2021-09-13 12:02:44,779 -   DEBUG -                         lib.py: 129 - Command ['irm', '-f', 'tmpml8KccirodsResc'] returned with code 0.
> 2021-09-13 12:02:44,779 -    INFO -                 setup_irods.py: 154 - Success.
> 2021-09-13 12:02:44,780 -    INFO -                 setup_irods.py: 134 - 
> +--------------------------------+
> | iRODS is installed and running |
> +--------------------------------+
> 

The irods installer tested iput, and it returned okay.

This is the content of /var/lib/irods/log/rodsLog.2021.09.11 for my iput:

> Sep 13 14:08:48 pid:22907 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> Sep 13 14:08:48 pid:22907 NOTICE: ssl_verify_callback: problem with certificate at depth: 0
> Sep 13 14:08:48 pid:22907 NOTICE: ssl_verify_callback:   issuer = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:48 pid:22907 NOTICE: ssl_verify_callback:   subject = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:48 pid:22907 NOTICE: ssl_verify_callback:   err 18:self signed certificate
> Sep 13 14:08:48 pid:22918 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> Sep 13 14:08:49 pid:22918 NOTICE: ssl_verify_callback: problem with certificate at depth: 0
> Sep 13 14:08:49 pid:22918 NOTICE: ssl_verify_callback:   issuer = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:49 pid:22918 NOTICE: ssl_verify_callback:   subject = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:49 pid:22918 NOTICE: ssl_verify_callback:   err 18:self signed certificate
> Sep 13 14:08:49 pid:22926 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> Sep 13 14:08:49 pid:22934 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> Sep 13 14:08:49 pid:22926 NOTICE: ssl_verify_callback: problem with certificate at depth: 0
> Sep 13 14:08:49 pid:22926 NOTICE: ssl_verify_callback:   issuer = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:49 pid:22926 NOTICE: ssl_verify_callback:   subject = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:49 pid:22926 NOTICE: ssl_verify_callback:   err 18:self signed certificate
> Sep 13 14:08:49 pid:22935 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> Sep 13 14:08:50 pid:22935 NOTICE: ssl_verify_callback: problem with certificate at depth: 0
> Sep 13 14:08:50 pid:22935 NOTICE: ssl_verify_callback:   issuer = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:50 pid:22935 NOTICE: ssl_verify_callback:   subject = /C=NL/L=Utrecht/O=Utrecht University/CN=test-yoda
> Sep 13 14:08:50 pid:22935 NOTICE: ssl_verify_callback:   err 18:self signed certificate
> Sep 13 14:08:50 pid:22950 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> Sep 13 14:08:50 pid:22950 remote addresses: IP ERROR: read_client_server_negotiation_message :: received error message -9000
> Sep 13 14:08:50 pid:22950 remote addresses: IP ERROR: _rcConnect: connectToRhost error, server on combined.test-yoda.FQDN:1247 is probably down status = -9000 SYS_EXCEED_CONNE
> CT_CNT
> Sep 13 14:08:50 pid:22950 remote addresses: IP ERROR: remoteFilePut - svrToSvrConnect failed -9000
> Sep 13 14:08:50 pid:22935 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22926 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22918 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22907 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22894 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22886 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22878 NOTICE: remoteFilePut: rcFilePut failed for /var/lib/irods/Vault1_1/home/rods/test.txt
> Sep 13 14:08:50 pid:22950 remote addresses: IP ERROR: read_client_server_negotiation_message :: received error message -9000
> Sep 13 14:08:50 pid:22950 remote addresses: IP ERROR: _rcConnect: connectToRhost error, server on combined.test-yoda.FQDN:1247 is probably down status = -9000 SYS_EXCEED_CONNE
> CT_CNT
> Sep 13 14:08:50 pid:22878 remote addresses: IP ERROR: _rsDataObjClose - getSizeInVault failed [-9000]
> Sep 13 14:08:50 pid:22958 NOTICE: writeLine: inString = Agent process started for puser=rods and cuser=rods from IP
> 



The iRODS troubleshooting page (https://docs.irods.org/4.1.6/manual/troubleshooting/) says that "-9000 SYS_EXCEED_CONNECT_CNT" often indicates problems with name resolving or routing. I do have the corresponding DNS records:

> nslookup combined.test-yoda.FQDN
> Server:		127.0.0.53
> Address:	127.0.0.53#53
> 
> Non-authoritative answer:
> combined.test-yoda.FQDN	canonical name = test-yoda.FQDN.
> Name:	test-yoda.FQDN
> Address: IP
> 

Am i missing something?

The first lines in /var/lib/irods/log/rodsLog.2021.09.11 show this:

> Sep 13 15:24:36 pid:15750 remote addresses: IP ERROR: [-]	/irods/server/core/src/rsApiHandler.cpp:540:int readAndProcClientMsg(rsComm_t *, int) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 	[-]	/irods/lib/core/src/sockComm.cpp:201:irods::error readMsgHeader(irods::network_object_ptr, msgHeader_t *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 		[-]	/irods/plugins/network/tcp/libtcp.cpp:194:irods::error tcp_read_msg_header(irods::plugin_context &, void *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [only read [0] of [4]]
> 
> Sep 13 15:24:36 pid:15750 remote addresses: IP ERROR: Agent [15750] exiting with status = -4000
> Sep 13 15:24:36 pid:13869  ERROR: Agent process [15750] exited with status [96]
> Sep 13 15:25:10 pid:17516 remote addresses: IP ERROR: [-]	/irods/server/core/src/rsApiHandler.cpp:540:int readAndProcClientMsg(rsComm_t *, int) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 	[-]	/irods/lib/core/src/sockComm.cpp:201:irods::error readMsgHeader(irods::network_object_ptr, msgHeader_t *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 		[-]	/irods/plugins/network/tcp/libtcp.cpp:194:irods::error tcp_read_msg_header(irods::plugin_context &, void *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [only read [0] of [4]]
> 
> Sep 13 15:25:10 pid:17516 remote addresses: IP ERROR: Agent [17516] exiting with status = -4000
> Sep 13 15:25:10 pid:13869  ERROR: Agent process [17516] exited with status [96]
> Sep 13 15:25:11 pid:17552 remote addresses: IP ERROR: [-]	/irods/server/core/src/rsApiHandler.cpp:540:int readAndProcClientMsg(rsComm_t *, int) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 	[-]	/irods/lib/core/src/sockComm.cpp:201:irods::error readMsgHeader(irods::network_object_ptr, msgHeader_t *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 		[-]	/irods/plugins/network/tcp/libtcp.cpp:194:irods::error tcp_read_msg_header(irods::plugin_context &, void *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [only read [0] of [4]]
> 
> Sep 13 15:25:11 pid:17552 remote addresses: IP ERROR: Agent [17552] exiting with status = -4000
> Sep 13 15:25:11 pid:13869  ERROR: Agent process [17552] exited with status [96]
> Sep 13 15:25:11 pid:17585 NOTICE: rsGeneralAdmin add child "dev001_1{}" to resource "dev001_p1"
> Sep 13 15:25:12 pid:17620 NOTICE: rsGeneralAdmin add child "dev001_2{}" to resource "dev001_p2"
> Sep 13 15:25:13 pid:17653 NOTICE: rsGeneralAdmin add child "dev001_p1{}" to resource "dev001"
> Sep 13 15:25:13 pid:17657 NOTICE: rsGeneralAdmin add child "dev001_p2{}" to resource "dev001"
> Sep 13 15:25:13 pid:17690 NOTICE: rsGeneralAdmin add child "dev001{}" to resource "irodsResc"
> Sep 13 15:25:14 pid:17725 remote addresses: IP ERROR: [-]	/irods/server/core/src/rsApiHandler.cpp:540:int readAndProcClientMsg(rsComm_t *, int) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 	[-]	/irods/lib/core/src/sockComm.cpp:201:irods::error readMsgHeader(irods::network_object_ptr, msgHeader_t *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [failed to call 'read header']
> 		[-]	/irods/plugins/network/tcp/libtcp.cpp:194:irods::error tcp_read_msg_header(irods::plugin_context &, void *, struct timeval *) :  status [SYS_HEADER_READ_LEN_ERR]  errno [] -- message [only read [0] of [4]]
> 
> Sep 13 15:25:14 pid:17725 remote addresses: IP ERROR: Agent [17725] exiting with status = -4000
> Sep 13 15:25:14 pid:13869  ERROR: Agent process [17725] exited with status [96]
> Sep 13 15:25:15 pid:17758 NOTICE: rsGeneralAdmin add child "dev002_1{}" to resource "dev002_p1"
> Sep 13 15:25:15 pid:17793 NOTICE: rsGeneralAdmin add child "dev002_p1{}" to resource "dev002"
> Sep 13 15:25:16 pid:17827 NOTICE: rsGeneralAdmin add child "dev002{}" to resource "irodsRescRepl"
> Sep 13 15:26:16 pid:20967 remote addresses: IP ERROR: caught python exception: Traceback (most recent call last):
>   File "/etc/irods/core.py", line 1, in <module>
>     from rules_uu import *
> ImportError: No module named rules_uu
> Sep 13 15:26:16 pid:20967 remote addresses: IP ERROR: caught python exception: Traceback (most recent call last):
>   File "/etc/irods/core.py", line 1, in <module>
>     from rules_uu import *
> ImportError: No module named rules_uu
> Sep 13 15:26:16 pid:20967 remote addresses: IP ERROR: caught python exception: Traceback (most recent call last):
>   File "/etc/irods/core.py", line 1, in <module>
>     from rules_uu import *
> ImportError: No module named rules_uu
> Sep 13 15:26:16 pid:20967 remote addresses: IP ERROR: executeRuleAction Failed for msiAclPolicy status = -1828000 RULE_ENGINE_ERROR
> Sep 13 15:26:16 pid:20967 NOTICE: executeRuleBody: Microservice or Action msiAclPolicy Failed with status -1828000
> Sep 13 15:26:16 pid:20967 NOTICE: execRuleNodeRes: applyRule Failed: acAclPolicy with status -1828000
> Sep 13 15:26:16 pid:20967 DEBUG: 
> Caught signal [11]. Dumping stacktrace and exiting

At this point, the remote connection seemed to be working. Is that supposed to happen?  Could that be the reason for my troube with irods? And if yes, how do I fix this?

Kind regards,

Gudrun

**Comment by stsnel** [comment]
Hi Gudrun,

What is the value of configuration parameter `irods_ssl_verify_server`? If the environment has both a self-signed certificate and iRODS certificate validation is enabled (i.e. playbook parameter `irods_ssl_verify_server` is set to its default value of `cert`), this would result in certificate validation errors. 

If you want to create a production(-like) environment, you could install a CA-signed certificate. You can find an example script for installing certificates on Yoda here: https://gist.github.com/stsnel/62c1e9147299ebcdc68072239338701c . Alternatively, you could temporarily switch `irods_ssl_verify_server` to `none`, which should cause iRODS to accept the self-signed certificate. 

I'm not sure about the cause of the ruleset error. Can you check whether it still occurs after either disabling certificate validation or installing a CA-signed certificate?

Best,
Sietse

**Comment by kalylian** [comment]
Hi,

`irods_ssl_verify_server` already is set to `none` in irods_environment.json:

> grep irods_ssl_verify_server irods_environment.json
>     "irods_ssl_verify_server": "none", 

`irods_ssl_verify_server` in group_vars is also set to `none`:
> grep irods_ssl_verify_server yoda/group_vars/yoda.yml 
> irods_ssl_verify_server: none              # Verify TLS certificate, use 'cert' for acceptance and production

Setting it to `cert` didn't work for me. The self-signed cert doesn't pass the validation - as expected -  and I don't now how to pass my own CA-signed cert file to the ansible script. I guess I need to set up everything with `none`, then replace the cert, then change the parameter..?

Also also, I replaced the self-signed certificate with one signed by our CA which included all FQDN defined in group_vars/yoda.yml as alternate name, which made the ssl notice in the logs go away but the -9000 error persisted.
Speaking of certs, is there a reason why `/etc/irods/localhost.key` is given permissions 644 and `root` ownership by ansible? 

Kind regards,

Gudrun

**Comment by lwesterhof** [comment]
Hi Gudrun,

Did you start with a new VM after you switched branches? Because the errors look like the deployment was done over the previous install.

Best regards,
Lazlo

**Comment by kalylian** [comment]
Hi,

I have a VM with a snapshot from before running ansible and I'm rolling back whenever I feel like I've thoroughly destroyed the VM. Which happens regularly. Still, you were close, it's a hostname issue which could have been because of rewriting files. It actually is because of me being unfamiliar with ansible. I followed the example of `allinone/hosts` when setting up my `hosts`-file, and I didn't interpret `combined` in that file as an FQDN. I used a shortener. Appearently, that resulted in `inventory_hostname` being set to said shortener. Which is what `hostname` is being set to in `roles/common/tasks/hostname.yml`.
Basically, my server had its hostname changed to `test-yoda`, which very obviously isn't an FQDN and therefore doesn't work with iRODS. I replaced all instances of `test-yoda` in `hosts` with the full FQDN. The playbook now ends with this:

> RUNNING HANDLER [apache : Restart firewall] ************************************************************************************************
> changed: [test-yoda.FQDN]
> 
> PLAY RECAP *********************************************************************************************************************************
> localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
> test-yoda.FQDN : ok=326  changed=184  unreachable=0    failed=0    skipped=49   rescued=0    ignored=11  
> 
> Playbook run took 0 days, 0 hours, 13 minutes, 25 seconds

I think this means everything went fine. Next step, setting up authentication. Thanks a lot for your time and help :)

Kind regards,

Gudrun
